File: Divide-and-Conquer Searching/Binary Search/Iterative Binary Search/IterativeBinarySearch.js
function iterativeBinarySearch(arr, target) {
    let left = 0;
    let right = arr.length - 1;

    while (left <= right) {
        const mid = left + Math.floor((right - left) / 2);

        if (arr[mid] === target) {
            return mid;
        }
        else if (arr[mid] < target) {
            left = mid + 1;
        }
        else {
            right = mid - 1;
        }
    }

    return -1;
}

const arr = [5, 12, 18, 25, 31, 42, 56];
const target = 31;

const index = iterativeBinarySearch(arr, target);

if (index !== -1) {
    console.log(
        `Element ${target} found at index ${index}`
    );
} else {
    console.log(
        `Element ${target} not found`
    );
}

File: Divide-and-Conquer Searching/Binary Search/Iterative Binary Search/IterativeBinarySearch.py
def iterative_binary_search(arr, target):
    left = 0
    right = len(arr) - 1

    while left <= right:
        mid = left + (right - left) // 2

        if arr[mid] == target:
            return mid

        elif arr[mid] < target:
            left = mid + 1

        else:
            right = mid - 1

    return -1


def main():
    arr = [5, 12, 18, 25, 31, 42, 56]
    target = 31

    index = iterative_binary_search(arr, target)

    if index != -1:
        print(f"Element {target} found at index {index}")
    else:
        print(f"Element {target} not found")


if __name__ == "__main__":
    main()
